src/ompl/base/StateSampler.h

```diff
@@ -76,6 +76,26 @@ namespace ompl
             {
             }
 
+            /** \brief Cast this instance to a desired type. */
+            template<class T>
+            T* as()
+            {
+                /** \brief Make sure the type we are casting to is indeed a planner */
+                BOOST_CONCEPT_ASSERT((boost::Convertible<T*, StateSampler*>));
+
+                return static_cast<T*>(this);
+            }
+
+            /** \brief Cast this instance to a desired type. */
+            template<class T>
+            const T* as() const
+            {
+                /** \brief Make sure the type we are casting to is indeed a Planner */
+                BOOST_CONCEPT_ASSERT((boost::Convertible<T*, StateSampler*>));
+
+                return static_cast<const T*>(this);
+            }
+
```

Comment:
Can we remove these cast operations? My understanding is that we will not need to cast things any more?